### PR TITLE
Fix Context::getTranslator() cache

### PR DESCRIPTION
**Pull request by rGaillard**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | 1.7.2.x
| Description?  | In the Context::getTranslator() method, the current translator is cached even if there is no translation file available. This leads to several problems like the translations of the fixtures with a CLI install.
| Type?         | bug fix 
| Category?     | CO
| BC breaks?    | no
| Deprecations? | no
| How to test?  | try to install PrestaShop in CLI in a language different of english, fixtures are now translated

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/8380)

### CODE REVIEW

**Comment by eternoendless** [comment]
:lgtm:

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/prestashop/prestashop/8380#-:-KvCwT9NCkhbn5PDebGW:bnfp4nl)*

**Review by eternoendless — APPROVED** [review]

**Comment by eternoendless** [comment]
@Quetzacoalt91 what do you think?

File: classes/Context.php [hunk 1]
```diff
@@ -348,32 +348,35 @@ public function getTranslator()
...
 
         // However, in some case, even empty catalog were stored in the cache and then used as-is.
         // For this one, we drop the cache and try to regenerate it.
-        $cache_file = Finder::create()
-            ->files()
-            ->in($cacheDir)
-            ->depth('==0')
-            ->name('*.'.$this->language->locale.'.*');
-        (new Filesystem())->remove($cache_file);
+        if (is_dir($cacheDir)) {
```
Comment (Quetzacoalt91) [inline classes/Context.php:362]: Is there a case where this folder does not exist?

Comment (rGaillard) [reply classes/Context.php:362]: When you are clearing your cache.

**Review by Quetzacoalt91 — APPROVED** [review]

**Pull request merged**